src/thor/triplegbuilder_utils.h

```diff
@@ -486,6 +489,9 @@ void AddIntersectingEdges(const AttributesController& controller,
   if (node->transition_count() > 0) {
     const NodeTransition* trans = start_tile->transition(node->transition_index());
     for (uint32_t i = 0; i < node->transition_count(); ++i, ++trans) {
+      // Skip nodes from lower levels if the corresponding flag is enabled
+      if (!allow_downwards_transitions && !trans->up())
+        continue;
```

Comment:
unfortunately we cant do this here. i think what we really want is to not call `AddIntersectingEdges` at all for nodes in the route which are between edges that were recovered from a shortcut. This will match the current implementation and not send confusing looking intersections to odin.